pgjdbc/src/test/java/org/postgresql/test/jdbc2/BinaryTest.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.test.jdbc2;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.postgresql.core.Oid;
+import org.postgresql.core.QueryExecutor;
+import org.postgresql.jdbc.PgConnection;
+import org.postgresql.test.TestUtil;
+import org.postgresql.util.PGBinaryObject;
+import org.postgresql.util.PGobject;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * TestCase to test handling of binary types.
+ */
+public class BinaryTest {
+  private Connection con;
+
+  /**
+   * Set up the fixture for this testcase: the tables for this test.
+   *
+   * @throws SQLException if a database error occurs
+   */
+  @Before
+  public void setUp() throws SQLException {
+    con = TestUtil.openDB();
+
+    TestUtil.createTable(con, "test_binary", "id integer,name text,geom point");
```

Comment:
creating and dropping tables should probably be in `BeforeClass` (or `BeforeAll` in junit5)